src/compiler/transformers/classFields.ts

```diff
@@ -9,20 +9,61 @@ namespace ts {
     }
 
     const enum PrivateIdentifierPlacement {
-        InstanceField
+        InstanceField,
+        InstanceMethod,
+        InstanceGetterOnly,
+        InstanceSetterOnly,
+        InstanceGetterAndSetter
     }
 
-    type PrivateIdentifierInfo = PrivateIdentifierInstanceField;
+    type PrivateIdentifierInfo =
+        | PrivateIdentifierInstanceField
+        | PrivateIdentifierInstanceMethod
+        | PrivateIdentifierInstanceGetterOnly

```

Comment:
Would it be simpler to have a single `PrivateIdentifierInstanceAccessor` which has optional `getterName` and `setterName` properties?

Reply:
I had that initially - it's pretty much the same complexity. This makes `addPrivateIdentifierToEnvironment` logic a little more complicated but then identifier access and assignment transforms can use `placement` in a unified way. If we had a single placement, `addPrivateIdentifierToEnvironment` would be a bit simpler but transforms would need to account of optionality.